src/event_batch.rs

```diff
@@ -30,4 +36,100 @@ impl EventBatch {
             data: json!(self.events),
         }
     }
+
+    /// Updates the events `stm` field in batch with the current time.
+    pub fn update_event_stm(&mut self) -> Result<(), Error> {
+        let since_the_epoch =
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .map_err(|e: SystemTimeError| {
+                    Error::BuilderError(format!("Failed to get current time: {}", e.to_string()))
+                })?;
+
+        for event in self.events.iter_mut() {
+            event.stm = since_the_epoch.as_millis().to_string();
+        }
+
+        Ok(())
+    }
+
+    ///
+    pub fn update_delay(&mut self) {
+        let max_event_delay_time = Duration::from_secs(10);
+
+        self.delay = match self.delay {
+            Some(delay) => {
+                // 2 + random number between 0 and 1
+                let delay_mul = rand::thread_rng().gen_range(2.0..=3.0);
```

Comment:
Nit: I think that the Java tracker generates values in the range 1 to 3, [see here](https://github.com/snowplow/snowplow-java-tracker/blob/master/src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/BatchEmitter.java#L504). Not a big deal, but might be nicer to be consistent.